rmw_zenoh_cpp/src/detail/graph_cache.cpp

```diff
@@ -41,170 +41,284 @@ using Entity = liveliness::Entity;
 using EntityType = liveliness::EntityType;
 
 ///=============================================================================
-TopicStats::TopicStats(std::size_t pub_count, std::size_t sub_count)
-: pub_count_(pub_count),
-  sub_count_(sub_count)
+TopicData::TopicData(
+  liveliness::TopicInfo info,
+  std::unordered_set<liveliness::Entity> pubs,
+  std::unordered_set<liveliness::Entity> subs)
+: info_(std::move(info)),
+  pubs_(std::move(pubs)),
+  subs_(std::move(subs))
```

Comment:
This is probably me just not know C++, but the ownership semantics are pretty unclear to me here.  Is `TopicData` taking ownership of these objects?  If so, why are they being copied in to begin with?  If `TopicData` is not taking ownership, what is the lifetime of these objects?  Would it be clearer to use a `std::unique_ptr` here?